### PR TITLE
Add logging for teacher and ?researchLogin

### LINKED ISSUE #1425: URL option to ?logout
If you are already logged in as someone else, and you want to get back to logged out status.

### PR DESCRIPTION
**Pull request by houshuang**
- Names all teacher log types with teacher.*
- Adds ?researchLogin (instead of ?login) for mode where teacher actions (other than next.activity which affects session state) are not logged. The username is shown with a start next to it. This should survive reloads.
- Currently logging: switching to/from dashboard, switching between dashboard activities and sub-dashboards, switching in/out of orchestration view, pausing. 
- Adds ?reset=true, which prevents logging in with stored login (useful to test the studentlist login, when you are already logged in as a user)

Sample of logging results:
screen shot 2018-10-03 at 10 54 58

Closes #1425

### CODE REVIEW

**Comment by jkolsen** [comment]
This looks good. I had a couple of questions:

What is the difference between switchActivityDashboard and switchDashboardActivity?
Can we tell when the teacher goes into the projector view and when they are on the projector view tab?

**Comment by houshuang** [comment]
Yeah the naming could be better. Basically switchDashboardActivity would be choosing a new activity, like Quiz. switchActivityDashboard is switching between different sub-dashboards (like for Quiz, progress, coordinates, justification etc).

You see teacherEnteredOrchestrationView and teacherLeftOrchestrationView. Note that this does not differentiate between leaving the orchestration tab and going to for example graph editor, or switching the session you're controlling - I don't know if this difference matters.

Not sure if I captured your meaning here:
> Can we tell when the teacher goes into the projector view and when they are on the projector view tab?
?

File: frog/imports/ui/TeacherView/utils/buttonUtils.js [hunk 1]
```diff
@@ -182,9 +189,27 @@ export const SessionUtilsButtonsModel = (
...
+    }
+  },
+  open3: {
+    button: {
+      onClick: () => {
+        ['Chen Li', 'Joanna', 'Marius'].forEach(x =>
+          window.open(`/${session.slug}?debugLogin=${x}`, uuid())
+        );
+      },
+      text: 'Open 3 student windows'
```
Comment (lfaucon) [inline frog/imports/ui/TeacherView/utils/buttonUtils.js:205]: For me the open3 button opens only one student called ChenLi

**Review by lfaucon — APPROVED** [review]
Everything seems to work very well for me

File: frog/imports/ui/App/TopBar.js [hunk 2]
```diff
@@ -90,7 +91,8 @@ class LogoutMenu extends React.Component<*, *> {
             onClick={() => {
               sessionStorage.removeItem('frog.sessionToken');
               Meteor.logout();
-              window.location.assign('/');
+              this.props.history.push('/');
+              window.notReady();
```
Comment (lfaucon) [inline frog/imports/ui/App/TopBar.js:95]: What does this do?

**Pull request merged**